src/FunctionApps/SilverTransforms/Transform/transforms.py

```diff
@@ -0,0 +1,67 @@
+import pymongo
+import smartystreets_python_sdk.us_street
+
+from Transform.geo import cached_geocode
+
+
+def transform_name(raw: str) -> str:
+    """trim spaces, capitalize, etc"""
+    raw = [x for x in raw if not x.isnumeric()]
+    raw = "".join(raw)
+    raw = raw.upper()
+    raw = raw.strip()
+    return raw
+
+
+def transform_phone(raw: str) -> str:
+    """Make sure it's 10 digits, remove everything else"""
+    raw = [x for x in raw if x.isnumeric()]
+    raw = "".join(raw)
+    if len(raw) != 10:
+        raw = None
+    return raw
+
+
+def transform_record(
+    cache: pymongo.collection.Collection,
+    client: smartystreets_python_sdk.us_street.Client,
+    patient: dict,
+) -> dict:
+
+    # Transform names
+    for i, name in enumerate(patient["name"]):
+        raw_last_name = name["family"]
+        patient["name"][i]["family"] = transform_name(raw_last_name)
```

Comment:
What's the structure of the `patient` dict? Is this supposed to be a dictionary containing _all_ patients (i.e. the first level of field down is a property, then the next field is patient `n` out of `num_patients`)? Based on the variable name I thought it would be a single patient so a little unclear on the structure of this object.

Reply:
It is a single patient, but a single patient may have more than one name (name changes). The FHIR patient resource stores first and last name in lists to allow for this. Most of the the time these lists will only have one element. We could just alway pick the most recent name (first element), which may be what we do for linkage, but it seemed easy enough to just standardize all names while interacting with the record.